src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -2146,42 +2146,51 @@ public boolean getCellSelectionEnabled() {
-     */
-    public void selectAll() {
-        // If I'm currently editing, then I should stop editing
-        if (isEditing()) {
-            removeEditor();
-        }
-        if (getRowCount() > 0 && getColumnCount() > 0) {
-            int oldLead;
-            int oldAnchor;
-            ListSelectionModel selModel;
+    private void selectRows(ListSelectionModel selModel, int rowCount) {
+        selModel.setValueIsAdjusting(true);
+        int oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
+        int oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
 
-            selModel = selectionModel;
-            selModel.setValueIsAdjusting(true);
-            oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), true);
-            oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), true);
+        setRowSelectionInterval(0, rowCount - 1);
 
-            setRowSelectionInterval(0, getRowCount()-1);
+        // this is done to restore the anchor and lead
+        SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+
+        selModel.setValueIsAdjusting(false);
+    }
 
-            // this is done to restore the anchor and lead
-            SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+    private void selectColumns(TableColumnModel  columnModel, int columnCount) {
+        ListSelectionModel selModel = columnModel.getSelectionModel();
+        selModel.setValueIsAdjusting(true);
+        int oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), false);
+        int oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), false);
 
-            selModel.setValueIsAdjusting(false);
+        setColumnSelectionInterval(0, columnCount - 1);
 
-            selModel = columnModel.getSelectionModel();
-            selModel.setValueIsAdjusting(true);
-            oldLead = getAdjustedIndex(selModel.getLeadSelectionIndex(), false);
-            oldAnchor = getAdjustedIndex(selModel.getAnchorSelectionIndex(), false);
+        // this is done to restore the anchor and lead
+        SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
 
-            setColumnSelectionInterval(0, getColumnCount()-1);
+        selModel.setValueIsAdjusting(false);
+    }
 
-            // this is done to restore the anchor and lead
-            SwingUtilities2.setLeadAnchorWithoutSelection(selModel, oldLead, oldAnchor);
+    /**
+     *  Selects all rows, columns, and cells in the table.
+     */
+    public void selectAll() {
+        int rowCount = getRowCount();
+        int columnCount = getColumnCount();
 
-            selModel.setValueIsAdjusting(false);
+        // If I'm currently editing, then I should stop editing
+        if (isEditing()) {
+            removeEditor();
+        }
+        if (rowCount > 0 && columnCount > 0) {
+            selectRows(selectionModel, rowCount);
```

Comment:
Any reason for passing selectionModel(s) as parameters ? Anyhow they are class variables and can be used directly right ?

Reply:
Yes, good catch..we need to use local variable as done in old code and not modify the class variables..Modified to store the instance variable in local and use that..

Reply:
[Here](https://github.com/openjdk/jdk/blob/577ede73d8e916bac9050d3bee80d2f18cc833a7/src/java.desktop/share/classes/javax/swing/JTable.java#L5629)

Reply:
I didn't meant that, I meant the other way. This would still modify the class variable right? And why do you want to use local variable ? Since you are getting selectionIndex, I don't think local variable would be helpful here.

Reply:
THis is the way it was done previously where also the action was done on local var..please see previous code....the selectionIndex was obtained from selectionModel class var..

Reply:
I meant w.r.t modifications applied to class and local reference. If both are pointing to same variable you can minimise passing it as parameter in methods selectColumns/Rows. Just a suggestion.

Reply:
Thank you for your suggestion but I will like to keep it this way as it was done in previous code as even though they are pointing to same variable, directly modifying class instance variable is fraught with danger so its better to act on local var as it was done previously..

Reply:
Modified to remove the param and made it more ocalised.